Catalogues of events and Event details buttons

DFMM11 opened the pull request:
This PR has been done on top of PR #18 and introduces the following:
- Page for a list of events the authenticated user-created
- Page for a list of available events (the request is temporary since there is no server request for it and used the default get events)
- Page for a list of joined events (has no current connections since it is not getting anything since there is no backend request)
- Join button appears when the user is not the owner of the event and is not participating (no logic since I don't know the documentation for the join operation, but everything is to work)
- Leave button appears when the user is not the event owner and is participating in the event. When clicked, a confirmation dialog appears.
- Edit Event and Delete Event buttons appear when the authenticated user is the owner of the event. The logic is not implemented.
- Interface (IIterableObjects) and prefab for lists of objects to prepare for listings of users
- UI component ConfirmationDialog to use in dialogs for confirmation purposes.

![image](https://github.com/DMindek/GroupUP/assets/64738080/f15d4dcd-2733-43d3-9747-7fe7e6ce7079)

All the event listings have this format, and the buttons have been incorporated in the EventDetails like the other button was done.

DFMM11 committed:
fe/feature: addition of list of serializables

DFMM11 committed:
Merge branch 'feature/login-frontend' into feature/view_user_frontend

DFMM11 committed:
fe/feat: Creation of User Owned events page and business logic.

DFMM11 committed:
fix

DFMM11 committed:
Merge branch 'development' into feature/catalogue_of_own_events_frontend

DFMM11 committed:
feat/fe: Catalogue done. Missing fix loading error

DFMM11 committed:
fe/fix: page is fixed... dumb viewmodel

DFMM11 committed:
Merge branch 'feature/view_event_details-frontend' into feature/catalogue_of_own_events_frontend

DFMM11 committed:
fe/feature: available events complete

DFMM11 committed:
fe/feature: frontend join event UI and logic is done

DFMM11 committed:
fe/feature: List of joined events page is done

DFMM11 committed:
fe/feature: leave button and dialog done

DMindek reviewed (changes requested):
The code looks good, however I think we should change it to display the event start time instead of the Location in the upcoming events screen, like what we discussed during the meeting.

DFMM11 committed:
Merge branch 'development' into feature/catalogue_of_own_events_frontend

DFMM11 committed:
be/extra: creation of bat script to build and run the docker

DFMM11 committed:
fe/fix: changed location as second field of event to date

DFMM11 committed:
fe/fix: changed from date to startTime

DFMM11 commented:
> The code looks good, however I think we should change it to display the event start time instead of the Location in the upcoming events screen, like what we discussed during the meeting.

It is done according to what was requested!! Thanks for the review.
I also added a script to build/run the docker. To build the backend, run `docker_scripts.bat build` and to run it, run `docker_scripts.bat run`
To end, I fixed a null point error in profiles where location was not well defined.

DMindek reviewed (approved):
Good work, nice thinking with the script.
